### PR TITLE
Add general tab functionality for all users on user/page

### LINKED ISSUE #871: Add a "General" tab to the user details page
There is currently a "General" tab when viewing your own user details page, but not when viewing the user details page of other users.

I think it would be helpful to add a "General" tab that displays the user's public data.  Specifically, the user's UUID, whether the user is an admin, the user's active identity, and a history all of the user's previous identities.  

All of this data is already returned in the response for the user details api call `/user/[uuid]`.

### PR DESCRIPTION
**Pull request by camus-code**
Resolves #871

### CODE REVIEW

File: src/components/UserDetail/GeneralTab.js [hunk 1]
```diff
@@ -213,26 +214,34 @@ class GeneralTab extends React.Component {
...
+      pubkey,
+      pubkeyStatus,
+      showPastUserIdentities
+    } = this.state;
     const userHasActivePaywall =
       user && user.newuserpaywalladdress && user.newuserpaywallamount;
     const isUserPageOwner = loggedInAsUserId === user.id;
     const hasTheRescanResult =
       amountOfCreditsAddedOnRescan !== undefined && rescanUserId === user.id;
+    const isAdminOrTheUser = user && (isAdmin || loggedInAsUserId === user.id);
```
Comment (tiagoalvesdulce) [inline src/components/UserDetail/GeneralTab.js]: 👍

File: src/components/UserDetail/GeneralTab.js [hunk 2]
```diff
@@ -213,26 +214,34 @@ class GeneralTab extends React.Component {
...
-            {isAdmin ? (
-              <ButtonWithLoadingIcon
-                className="c-btn c-btn-primary button-small"
-                isLoading={isLoadingRescan}
-                onClick={() => onRescan(user.id)}
-                text="rescan"
-              />
-            ) : null}
-          </Field>
+          {isAdminOrTheUser ? (
```
Comment (tiagoalvesdulce) [inline src/components/UserDetail/GeneralTab.js]: if you aren't using an `else` clause, e.g `condition ? doSomething() : null`, there's no need to use the ternary operator, you can go with `condition && doSomething()` for simplicity.

File: src/components/UserDetail/GeneralTab.js [hunk 3]
```diff
@@ -524,7 +603,14 @@ class GeneralTab extends React.Component {
             </Field>
           </React.Fragment>
         )}
+        {isAdminOrTheUser ? <FieldSeparator /> : null}
```
Comment (tiagoalvesdulce) [inline src/components/UserDetail/GeneralTab.js]: Here you are doing something like: `condition ? doSomething() : null` and then `!condition ? doSomethingElse() : null`. Just do: `condition ? doSomething() : doSomethingElse()`

File: src/components/UserDetail/index.js [hunk 4]
```diff
@@ -20,17 +17,11 @@ class UserDetail extends Component {
...
     this.props.onFetchProposalsVoteStatus();
   }
-
-  componentDidUpdate(prevProps) {
-    const { loggedInAsUserId, editUserResponse, user, isAdmin } = this.props;
-    if (editUserResponse) {
+  componentDidUpdate() {
+    const { editUserResponse } = this.props;
+    if(editUserResponse) {
       window.location.reload();
```
Comment (tiagoalvesdulce) [inline src/components/UserDetail/index.js]: Why do we need this?

Comment (camus-code) [reply src/components/UserDetail/index.js]: Not too sure to be honest. It was in here beforehand and I couldn't exactly figure out the purpose so I was a little worried to remove it.

File: src/components/UserDetail/GeneralTab.js (hunk 3 above)
Comment (camus-code) [reply src/components/UserDetail/GeneralTab.js]: thanks for the suggestion mate. went through and took out all the unnecessary ternaries

**Comment by tiagoalvesdulce** [comment]
Depends on https://github.com/decred/politeia/pull/598

**Comment by tiagoalvesdulce** [comment]
There is an issue showing the active public key for the user when accessing the route logged out. Also, I think it shouldn't be `unpaid_user's Public Key` but `Active Public Key` instead. Another thing that might be related to this PR is that the number of proposals is not showing in the `Proposals` tab when it's 0.

screen shot 2018-11-14 at 20 12 41

**Comment by camus-code** [comment]
I think my last commit covered just about everything @tiagoalvesdulce . Ready for a look when you have a chance

**Comment by tiagoalvesdulce** [comment]
Looks good but the paddings are a bit messed up when looking another user's tab. Look:

screen shot 2018-11-20 at 09 53 51

**Pull request merged**